src/pudl/transform/eia.py

```diff
@@ -260,37 +258,44 @@ def _round_operating_date(
         cols_to_consit: a list of the columns to determine consistency.  This either the
             [entity_id] or the [entity_id, 'report_date'], depending on whether the
             entity is static or annual.
-        group_by_freq: Frequency to combine by ("M" for month, or "Y" for year)
 
     Returns:
         A dataframe with all of the entity ids. Some will have harvested records from
         the clean_df. Some will have NA values if no consistently reported records were
         found.
     """
-    # grab the dirty plant records, round and get a new consistency
-    op_df = dirty_df.assign(
-        operating_rounded=dirty_df[col].dt.to_period(group_by_freq).dt.to_timestamp()
+    # take the last reported date for each unharvested generator.
+    grouped = (
+        dirty_df.sort_values("report_date").groupby(by=entity_idx)[col].agg(["last"])
+    )
+
+    logger.info(
+        f"Taking the last generator operating date reported for {len(grouped)} generators."
     )
-    logger.debug(f"Dirty {col} records: {len(op_df)}")
+    op_df = dirty_df.merge(grouped, on=entity_idx)
+
+    op_df["last"] = op_df["last"].fillna(
+        op_df[col]
+    )  # If no value reported in last time slot, keep original.
+    op_df = op_df.drop(columns=col).rename(columns={"last": col})
+
     # Group all records within the same rounded time period and assign them the max
     # value within that time period.
-    op_df[col] = op_df.groupby(cols_to_consit + ["operating_rounded"])[col].transform(
-        "max"
-    )
     op_df["table"] = "special_case"
-    op_df = op_df.drop("operating_rounded", axis=1)
     op_df = occurrence_consistency(entity_idx, op_df, col, cols_to_consit)
     # grab the clean plants
     op_clean_df = clean_df.dropna()
     # find the new clean plant records by selecting the True consistent records
     op_df = op_df[op_df[f"{col}_is_consistent"]].drop_duplicates(subset=entity_idx)
-    logger.info(f"Clean {col} records: {len(op_df)}")
+    logger.info(f"Rescued dates for {col} records: {len(op_df)}")
     logger.info(
-        f"Rescued rounded {col} for the following units ({entity_idx}): "
+        f"Rescued last {col} for the following units ({entity_idx}): "
         f"{sorted(op_df[entity_idx].apply(lambda row: '_'.join(row.to_numpy().astype(str)), axis=1))}"
     )
     # add the newly cleaned records
     op_clean_df = pd.concat([op_clean_df, op_df])
+    # assert all generator operating dates are not null
+    assert len(op_clean_df[op_clean_df.generator_operating_date.isnull()]) == 0
```

Comment:
This is a minor non-blocking nit, but it feels more readable to use `any()`/`all()` for this kind of test. E.g.
```suggestion
    assert op_clean_df.generator_operating_date.notnull().all()
```